scripts/Packages/Package_BGA/bga.yaml

```diff
@@ -345,3 +345,16 @@ WLP-9_1.448x1.468mm_Layout3x3_P0.4mm_Ball0.27mm_Pad0.25mm:
   paste_margin: 0.000001
   layout_x: 3
   layout_y: 3
+BGA-320_15.0x15.0mm_Layout18x18_P0.8mm:
+  description: "BGA-320 (4 balls removed)"
+  size_source: "https://colognechip.com/docs/gatemate-ccgm1a-datasheet.pdf#page=48"
+  additional_tags: "GateMate FPGA"
+  body_size_x: 15
+  body_size_y: 15
+  overall_height: 1.24
+  pitch: 0.8
+  pad_diameter: 0.45
+  mask_margin: 0.05
```

Comment:
How did you deduct those parameters? I can't find them in the datasheet.

I would suggest to put this instead:
```yaml
  ball_type: 'collapsible'
  ball_diameter: 0.45
```
(that feature was just recently introduced in #592)

Omit the `mask_margin`, that way it is controlled by the DRC of the board. I don't see a recommendation from the manufacturer.

Reply:
> How did you deduct those parameters? I can't find them in the datasheet.

No, they are not in the datasheet, and, apparently, my initial submission is incorrect:
 -- general recommendations for 0.8mm pitch/0.45mm ball diameter FBGA packages
(based on documents from Intel, Xilinx, Lattice, and NXP) are NSMD pads
0.34..0.40 mm in diameter.  With the ipc_definitions.yaml we are going to have 0.35mm
pads, if I'm not misreading it.

> Since I am interested in that part as well I might just call the people at ColongeChip an ask.

I've sent the question to a support contact there.  Let's see if and when they
have something to answer.  Of course, you may call them too, it may be a faster
way to get the info.  Thanks.

> I would suggest to put this instead:
> 
> ```yaml
>   ball_type: 'collapsible'
>   ball_diameter: 0.45
> ```
> 
> (that feature was just recently introduced in #592)

Generally, makes sense and I've tried to implement the changes,
but, it doesn't work for me right away for a number of reasons:
 - I cannot find an example of its use in any other BGA package
 - "collapsible" vs "collapsable" -- ipc_definitions.yaml makes use of the later,
   while the correct spelling is the former (as you suggested above)
 - the bga.py script complains about missing pad_diameter:

```
~/Work/kicad/kicad-footprint-generator/scripts/Packages/Package_BGA$ python3 bga.py ccgm.yaml 
Building footprint for parameter set: BGA-320_15.0x15.0mm_Layout18x18_P0.8mm
Traceback (most recent call last):
  File "bga.py", line 232, in <module>
    generateFootprint(configuration, cmd_file[pkg], pkg)
  File "bga.py", line 174, in generateFootprint
    size=[fpParams["pad_diameter"], fpParams["pad_diameter"]],
KeyError: 'pad_diameter'
```
Perhaps, I've just missed some critical step to refer to the proper IPC file,
but I don't know how to do it, sorry.

> Omit the `mask_margin`, that way it is controlled by the DRC of the board. I don't see a recommendation from the manufacturer.

Agree.

Reply:
I just called CologneChip and taked to Mr. U.... (redacted for privacy reasons).

The important information is that the package and pinout will change slighlty. They are in the process of finishing the documentation for that. Once the up-to-date documentation is available they will provide it to us.
Thus, this PR and the symbol-PR will have to wait a few weeks.